extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -443,13 +443,22 @@ export class ProjectsController {
 	}
 
 	public async updateProjectForRoundTrip(project: Project) {
-		if (project.importedTargets.includes(constants.NetCoreTargets)) {
+		if (project.importedTargets.includes(constants.NetCoreTargets) && !project.containsSSDTOnlySystemDatabaseReferences()) {
 			return;
 		}
 
-		const result = await this.apiWrapper.showWarningMessage(constants.updateProjectForRoundTrip, constants.yesString, constants.noString);
-		if (result === constants.yesString) {
-			await project.updateProjectForRoundTrip();
+		if (!project.importedTargets.includes(constants.NetCoreTargets)) {
+			const result = await this.apiWrapper.showWarningMessage(constants.updateProjectForRoundTrip, constants.yesString, constants.noString);
+			if (result === constants.yesString) {
+				await project.updateProjectForRoundTrip();
+			}
+		}
+
+		if (project.containsSSDTOnlySystemDatabaseReferences()) {
+			const result = await this.apiWrapper.showWarningMessage(constants.updateProjectDatabaseReferencesForRoundTrip, constants.yesString, constants.noString);
+			if (result === constants.yesString) {
+				await project.updateSystemDatabaseReferencesInProjFile();
+			}
```

Comment:
What about collecting everything that needs updating first, then only asking the user once for everything?  I can't think of a reason a user might click yes for one but no for the other.

Reply:
makes sense. I updated the roundtrip message to include system db refs and so the other system db specific message will only show up if the project has already been updated with the .NET Core targets